<?xml version="1.0" encoding="UTF-8"?>
<ui version="4.0">
 <class>MainWindow</class>
 <widget class="QMainWindow" name="MainWindow">
  <property name="geometry">
   <rect>
    <x>0</x>
    <y>0</y>
    <width>1385</width>
    <height>961</height>
   </rect>
  </property>
  <property name="minimumSize">
   <size>
    <width>1147</width>
    <height>500</height>
   </size>
  </property>
  <property name="windowTitle">
   <string>MainWindow</string>
  </property>
  <widget class="QWidget" name="centralwidget">
   <widget class="QPushButton" name="uploadButton">
    <property name="geometry">
     <rect>
      <x>30</x>
      <y>840</y>
      <width>64</width>
      <height>64</height>
     </rect>
    </property>
    <property name="styleSheet">
     <string notr="true">QPushButton {
	background-color: #666;
	border-radius: 32px;
	border: 1px solid #333;
}</string>
    </property>
    <property name="text">
     <string/>
    </property>
    <property name="icon">
     <iconset resource="../resources.qrc">
      <normaloff>:/assets/icons/upload.svg</normaloff>:/assets/icons/upload.svg</iconset>
    </property>
    <property name="iconSize">
     <size>
      <width>24</width>
      <height>24</height>
     </size>
    </property>
   </widget>
   <widget class="Canvas" name="canvas">
    <property name="geometry">
     <rect>
      <x>0</x>
      <y>0</y>
      <width>1391</width>
      <height>941</height>
     </rect>
    </property>
    <property name="styleSheet">
     <string notr="true">openGLWidget {
	min-size: 100%;
	max-size: 100%;
}</string>
    </property>
   </widget>
   <widget class="QLabel" name="filepath_label">
    <property name="geometry">
     <rect>
      <x>110</x>
      <y>855</y>
      <width>671</width>
      <height>31</height>
     </rect>
    </property>
    <property name="font">
     <font>
      <pointsize>18</pointsize>
     </font>
    </property>
    <property name="styleSheet">
     <string notr="true">QLabel {
	color: white;
}</string>
    </property>
    <property name="text">
     <string>path / to / obj / file</string>
    </property>
   </widget>
   <widget class="QGroupBox" name="groupBox_4">
    <property name="geometry">
     <rect>
      <x>1160</x>
      <y>70</y>
      <width>200</width>
      <height>71</height>
     </rect>
    </property>
    <property name="layoutDirection">
     <enum>Qt::LeftToRight</enum>
    </property>
    <property name="styleSheet">
     <string notr="true">color:white;
background-color: rgba(255,255,255,0.1);</string>
    </property>
    <property name="title">
     <string>Projection</string>
    </property>
    <property name="alignment">
     <set>Qt::AlignLeading|Qt::AlignLeft|Qt::AlignVCenter</set>
    </property>
    <widget class="QPushButton" name="pushButton">
     <property name="geometry">
      <rect>
       <x>10</x>
       <y>30</y>
       <width>91</width>
       <height>31</height>
      </rect>
     </property>
     <property name="styleSheet">
      <string notr="true">color:white;</string>
     </property>
     <property name="text">
      <string>parallel</string>
     </property>
    </widget>
    <widget class="QPushButton" name="pushButton_2">
     <property name="geometry">
      <rect>
       <x>100</x>
       <y>30</y>
       <width>91</width>
       <height>31</height>
      </rect>
     </property>
     <property name="styleSheet">
      <string notr="true">color:white;</string>
     </property>
     <property name="text">
      <string>central</string>
     </property>
    </widget>
   </widget>
   <widget class="QTabWidget" name="tabWidget">
    <property name="geometry">
     <rect>
      <x>1140</x>
      <y>260</y>
      <width>241</width>
      <height>151</height>
     </rect>
    </property>
    <property name="layoutDirection">
     <enum>Qt::LeftToRight</enum>
    </property>
    <property name="tabPosition">
     <enum>QTabWidget::North</enum>
    </property>
    <property name="tabShape">
     <enum>QTabWidget::Triangular</enum>
    </property>
    <property name="currentIndex">
     <number>0</number>
    </property>
    <property name="elideMode">
     <enum>Qt::ElideLeft</enum>
    </property>
    <widget class="QWidget" name="tab">
     <attribute name="title">
      <string>Position</string>
     </attribute>
     <widget class="QGroupBox" name="groupBox">
      <property name="geometry">
       <rect>
        <x>20</x>
        <y>0</y>
        <width>200</width>
        <height>111</height>
       </rect>
      </property>
      <property name="layoutDirection">
       <enum>Qt::LeftToRight</enum>
      </property>
      <property name="styleSheet">
       <string notr="true">color:white;
background-color: rgba(255,255,255,0.1);</string>
      </property>
      <property name="title">
       <string/>
      </property>
      <property name="alignment">
       <set>Qt::AlignLeading|Qt::AlignLeft|Qt::AlignVCenter</set>
      </property>
      <widget class="QLabel" name="label_2">
       <property name="geometry">
        <rect>
         <x>10</x>
         <y>10</y>
         <width>21</width>
         <height>31</height>
        </rect>
       </property>
       <property name="styleSheet">
        <string notr="true">color:white;
background-color: none;</string>
       </property>
       <property name="text">
        <string>x</string>
       </property>
       <property name="alignment">
        <set>Qt::AlignRight|Qt::AlignTrailing|Qt::AlignVCenter</set>
       </property>
      </widget>
      <widget class="QLabel" name="label_3">
       <property name="geometry">
        <rect>
         <x>10</x>
         <y>40</y>
         <width>21</width>
         <height>31</height>
        </rect>
       </property>
       <property name="styleSheet">
        <string notr="true">color:white;
background-color: none;</string>
       </property>
       <property name="text">
        <string>y</string>
       </property>
       <property name="alignment">
        <set>Qt::AlignRight|Qt::AlignTrailing|Qt::AlignVCenter</set>
       </property>
      </widget>
      <widget class="QLabel" name="label_4">
       <property name="geometry">
        <rect>
         <x>10</x>
         <y>70</y>
         <width>21</width>
         <height>31</height>
        </rect>
       </property>
       <property name="styleSheet">
        <string notr="true">color:white;
background-color: none;</string>
       </property>
       <property name="text">
        <string>z</string>
       </property>
       <property name="alignment">
        <set>Qt::AlignRight|Qt::AlignTrailing|Qt::AlignVCenter</set>
       </property>
      </widget>
      <widget class="QDoubleSpinBox" name="position_x">
       <property name="geometry">
        <rect>
         <x>40</x>
         <y>10</y>
         <width>151</width>
         <height>31</height>
        </rect>
       </property>
       <property name="styleSheet">
        <string notr="true">color:white;</string>
       </property>
       <property name="minimum">
        <double>-99.989999999999995</double>
       </property>
      </widget>
      <widget class="QDoubleSpinBox" name="position_y">
       <property name="geometry">
        <rect>
         <x>40</x>
         <y>40</y>
         <width>151</width>
         <height>31</height>
        </rect>
       </property>
       <property name="styleSheet">
        <string notr="true">color:white;</string>
       </property>
       <property name="minimum">
        <double>-99.989999999999995</double>
       </property>
      </widget>
      <widget class="QDoubleSpinBox" name="position_z">
       <property name="geometry">
        <rect>
         <x>40</x>
         <y>70</y>
         <width>151</width>
         <height>31</height>
        </rect>
       </property>
       <property name="styleSheet">
        <string notr="true">color:white;</string>
       </property>
       <property name="minimum">
        <double>-99.989999999999995</double>
       </property>
      </widget>
     </widget>
    </widget>
    <widget class="QWidget" name="tab_2">
     <attribute name="title">
      <string>Rotation</string>
     </attribute>
     <widget class="QGroupBox" name="groupBox_2">
      <property name="geometry">
       <rect>
        <x>20</x>
        <y>0</y>
        <width>200</width>
        <height>111</height>
       </rect>
      </property>
      <property name="layoutDirection">
       <enum>Qt::LeftToRight</enum>
      </property>
      <property name="styleSheet">
       <string notr="true">color:white;
background-color: rgba(255,255,255,0.1);</string>
      </property>
      <property name="title">
       <string/>
      </property>
      <property name="alignment">
       <set>Qt::AlignLeading|Qt::AlignLeft|Qt::AlignVCenter</set>
      </property>
      <widget class="QLabel" name="label_5">
       <property name="geometry">
        <rect>
         <x>10</x>
         <y>10</y>
         <width>21</width>
         <height>31</height>
        </rect>
       </property>
       <property name="styleSheet">
        <string notr="true">color:white;
background-color: none;</string>
       </property>
       <property name="text">
        <string>x</string>
       </property>
       <property name="alignment">
        <set>Qt::AlignRight|Qt::AlignTrailing|Qt::AlignVCenter</set>
       </property>
      </widget>
      <widget class="QLabel" name="label_6">
       <property name="geometry">
        <rect>
         <x>10</x>
         <y>40</y>
         <width>21</width>
         <height>31</height>
        </rect>
       </property>
       <property name="styleSheet">
        <string notr="true">color:white;
background-color: none;</string>
       </property>
       <property name="text">
        <string>y</string>
       </property>
       <property name="alignment">
        <set>Qt::AlignRight|Qt::AlignTrailing|Qt::AlignVCenter</set>
       </property>
      </widget>
      <widget class="QLabel" name="label_7">
       <property name="geometry">
        <rect>
         <x>10</x>
         <y>70</y>
         <width>21</width>
         <height>31</height>
        </rect>
       </property>
       <property name="styleSheet">
        <string notr="true">color:white;
background-color: none;</string>
       </property>
       <property name="text">
        <string>z</string>
       </property>
       <property name="alignment">
        <set>Qt::AlignRight|Qt::AlignTrailing|Qt::AlignVCenter</set>
       </property>
      </widget>
      <widget class="QDoubleSpinBox" name="rotation_x">
       <property name="geometry">
        <rect>
         <x>40</x>
         <y>10</y>
         <width>151</width>
         <height>31</height>
        </rect>
       </property>
       <property name="styleSheet">
        <string notr="true">color:white;</string>
       </property>
       <property name="minimum">
        <double>-99.989999999999995</double>
       </property>
      </widget>
      <widget class="QDoubleSpinBox" name="rotation_y">
       <property name="geometry">
        <rect>
         <x>40</x>
         <y>40</y>
         <width>151</width>
         <height>31</height>
        </rect>
       </property>
       <property name="styleSheet">
        <string notr="true">color:white;</string>
       </property>
       <property name="minimum">
        <double>-99.989999999999995</double>
       </property>
      </widget>
      <widget class="QDoubleSpinBox" name="rotation_z">
       <property name="geometry">
        <rect>
         <x>40</x>
         <y>70</y>
         <width>151</width>
         <height>31</height>
        </rect>
       </property>
       <property name="styleSheet">
        <string notr="true">color:white;</string>
       </property>
       <property name="minimum">
        <double>-99.989999999999995</double>
       </property>
      </widget>
     </widget>
    </widget>
    <widget class="QWidget" name="tab_3">
     <attribute name="title">
      <string>Scale</string>
     </attribute>
     <widget class="QGroupBox" name="groupBox_3">
      <property name="geometry">
       <rect>
        <x>20</x>
        <y>0</y>
        <width>200</width>
        <height>111</height>
       </rect>
      </property>
      <property name="layoutDirection">
       <enum>Qt::LeftToRight</enum>
      </property>
      <property name="styleSheet">
       <string notr="true">color:white;
background-color: rgba(255,255,255,0.1);</string>
      </property>
      <property name="title">
       <string/>
      </property>
      <property name="alignment">
       <set>Qt::AlignLeading|Qt::AlignLeft|Qt::AlignVCenter</set>
      </property>
      <widget class="QLabel" name="label_10">
       <property name="geometry">
        <rect>
         <x>0</x>
         <y>40</y>
         <width>41</width>
         <height>31</height>
        </rect>
       </property>
       <property name="styleSheet">
        <string notr="true">color:white;
background-color: none;</string>
       </property>
       <property name="text">
        <string>scale</string>
       </property>
       <property name="alignment">
        <set>Qt::AlignRight|Qt::AlignTrailing|Qt::AlignVCenter</set>
       </property>
      </widget>
      <widget class="QDoubleSpinBox" name="scale_control">
       <property name="geometry">
        <rect>
         <x>50</x>
         <y>40</y>
         <width>141</width>
         <height>31</height>
        </rect>
       </property>
       <property name="styleSheet">
        <string notr="true">color:white;</string>
       </property>
      </widget>
     </widget>
    </widget>
   </widget>
   <widget class="QTabWidget" name="tabWidget_2">
    <property name="geometry">
     <rect>
      <x>1140</x>
      <y>420</y>
      <width>241</width>
      <height>221</height>
     </rect>
    </property>
    <property name="currentIndex">
     <number>0</number>
    </property>
    <widget class="QWidget" name="tab_4">
     <attribute name="title">
      <string>Edges</string>
     </attribute>
     <widget class="QGroupBox" name="groupBox_9">
      <property name="geometry">
       <rect>
        <x>20</x>
        <y>110</y>
        <width>201</width>
        <height>71</height>
       </rect>
      </property>
      <property name="styleSheet">
       <string notr="true">color:white;
background-color: rgba(255,255,255,0.1);</string>
      </property>
      <property name="title">
       <string>Thikness</string>
      </property>
      <widget class="QSlider" name="horizontalSlider">
       <property name="geometry">
        <rect>
         <x>10</x>
         <y>30</y>
         <width>151</width>
         <height>25</height>
        </rect>
       </property>
       <property name="styleSheet">
        <string notr="true">background-color:none;</string>
       </property>
       <property name="orientation">
        <enum>Qt::Horizontal</enum>
       </property>
      </widget>
     </widget>
     <widget class="QGroupBox" name="groupBox_6">
      <property name="geometry">
       <rect>
        <x>20</x>
        <y>0</y>
        <width>201</width>
        <height>101</height>
       </rect>
      </property>
      <property name="styleSheet">
       <string notr="true">color:white;
background-color: rgba(255,255,255,0.1);</string>
      </property>
      <property name="title">
       <string>Type</string>
      </property>
      <widget class="QRadioButton" name="radioButton">
       <property name="geometry">
        <rect>
         <x>10</x>
         <y>50</y>
         <width>99</width>
         <height>20</height>
        </rect>
       </property>
       <property name="styleSheet">
        <string notr="true">background-color:none;</string>
       </property>
       <property name="text">
        <string>Solid</string>
       </property>
      </widget>
      <widget class="QRadioButton" name="radioButton_2">
       <property name="geometry">
        <rect>
         <x>10</x>
         <y>70</y>
         <width>99</width>
         <height>20</height>
        </rect>
       </property>
       <property name="styleSheet">
        <string notr="true">background-color:none;</string>
       </property>
       <property name="text">
        <string>Dashed</string>
       </property>
      </widget>
      <widget class="QRadioButton" name="radioButton_6">
       <property name="geometry">
        <rect>
         <x>10</x>
         <y>30</y>
         <width>99</width>
         <height>20</height>
        </rect>
       </property>
       <property name="styleSheet">
        <string notr="true">background-color:none;</string>
       </property>
       <property name="text">
        <string>None</string>
       </property>
      </widget>
     </widget>
    </widget>
    <widget class="QWidget" name="tab_5">
     <attribute name="title">
      <string>Vertices</string>
     </attribute>
     <widget class="QGroupBox" name="groupBox_10">
      <property name="geometry">
       <rect>
        <x>20</x>
        <y>0</y>
        <width>201</width>
        <height>101</height>
       </rect>
      </property>
      <property name="styleSheet">
       <string notr="true">color:white;
background-color: rgba(255,255,255,0.1);</string>
      </property>
      <property name="title">
       <string>Display method</string>
      </property>
      <widget class="QRadioButton" name="radioButton_3">
       <property name="geometry">
        <rect>
         <x>10</x>
         <y>30</y>
         <width>99</width>
         <height>20</height>
        </rect>
       </property>
       <property name="styleSheet">
        <string notr="true">background-color:none;</string>
       </property>
       <property name="text">
        <string>None</string>
       </property>
      </widget>
      <widget class="QRadioButton" name="radioButton_4">
       <property name="geometry">
        <rect>
         <x>10</x>
         <y>50</y>
         <width>99</width>
         <height>20</height>
        </rect>
       </property>
       <property name="styleSheet">
        <string notr="true">background-color:none;</string>
       </property>
       <property name="text">
        <string>Circle</string>
       </property>
      </widget>
      <widget class="QRadioButton" name="radioButton_5">
       <property name="geometry">
        <rect>
         <x>10</x>
         <y>70</y>
         <width>99</width>
         <height>20</height>
        </rect>
       </property>
       <property name="styleSheet">
        <string notr="true">background-color:none;</string>
       </property>
       <property name="text">
        <string>Square</string>
       </property>
      </widget>
     </widget>
     <widget class="QGroupBox" name="groupBox_12">
      <property name="geometry">
       <rect>
        <x>20</x>
        <y>110</y>
        <width>201</width>
        <height>71</height>
       </rect>
      </property>
      <property name="styleSheet">
       <string notr="true">color:white;
background-color: rgba(255,255,255,0.1);</string>
      </property>
      <property name="title">
       <string>Size</string>
      </property>
      <widget class="QSlider" name="horizontalSlider_2">
       <property name="geometry">
        <rect>
         <x>10</x>
         <y>30</y>
         <width>181</width>
         <height>25</height>
        </rect>
       </property>
       <property name="styleSheet">
        <string notr="true">background-color:none;</string>
       </property>
       <property name="orientation">
        <enum>Qt::Horizontal</enum>
       </property>
      </widget>
     </widget>
    </widget>
   </widget>
   <widget class="QGroupBox" name="groupBox_13">
    <property name="geometry">
     <rect>
      <x>1160</x>
      <y>150</y>
      <width>200</width>
      <height>91</height>
     </rect>
    </property>
    <property name="styleSheet">
     <string notr="true">color:white;
background-color: rgba(255,255,255,0.1);</string>
    </property>
    <property name="title">
     <string>Colors</string>
    </property>
    <widget class="QPushButton" name="pushButton_6">
     <property name="geometry">
      <rect>
       <x>10</x>
       <y>30</y>
       <width>51</width>
       <height>51</height>
      </rect>
     </property>
     <property name="styleSheet">
      <string notr="true">color:white;</string>
     </property>
     <property name="text">
      <string/>
     </property>
    </widget>
    <widget class="QPushButton" name="pushButton_8">
     <property name="geometry">
      <rect>
       <x>70</x>
       <y>30</y>
       <width>51</width>
       <height>51</height>
      </rect>
     </property>
     <property name="styleSheet">
      <string notr="true">background-color:red;</string>
     </property>
     <property name="text">
      <string/>
     </property>
    </widget>
    <widget class="QPushButton" name="pushButton_9">
     <property name="geometry">
      <rect>
       <x>130</x>
       <y>30</y>
       <width>51</width>
       <height>51</height>
      </rect>
     </property>
     <property name="styleSheet">
      <string notr="true">background-color:cyan;</string>
     </property>
     <property name="text">
      <string/>
     </property>
    </widget>
   </widget>
   <widget class="QPushButton" name="pushButton_4">
    <property name="geometry">
     <rect>
      <x>1160</x>
      <y>10</y>
      <width>101</width>
      <height>51</height>
     </rect>
    </property>
    <property name="styleSheet">
     <string notr="true">color:white;</string>
    </property>
    <property name="text">
     <string/>
    </property>
    <property name="icon">
     <iconset resource="../resources.qrc">
      <normaloff>:/assets/icons/screenshot.svg</normaloff>:/assets/icons/screenshot.svg</iconset>
    </property>
    <property name="iconSize">
     <size>
      <width>24</width>
      <height>24</height>
     </size>
    </property>
   </widget>
   <widget class="QPushButton" name="pushButton_7">
    <property name="geometry">
     <rect>
      <x>1260</x>
      <y>10</y>
      <width>101</width>
      <height>51</height>
     </rect>
    </property>
    <property name="text">
     <string/>
    </property>
    <property name="icon">
     <iconset resource="../resources.qrc">
      <normaloff>:/assets/icons/screen_record.svg</normaloff>:/assets/icons/screen_record.svg</iconset>
    </property>
    <property name="iconSize">
     <size>
      <width>24</width>
      <height>24</height>
     </size>
    </property>
   </widget>
   <zorder>canvas</zorder>
   <zorder>uploadButton</zorder>
   <zorder>filepath_label</zorder>
   <zorder>groupBox_4</zorder>
   <zorder>tabWidget</zorder>
   <zorder>tabWidget_2</zorder>
   <zorder>groupBox_13</zorder>
   <zorder>pushButton_4</zorder>
   <zorder>pushButton_7</zorder>
  </widget>
  <widget class="QMenuBar" name="menubar">
   <property name="geometry">
    <rect>
     <x>0</x>
     <y>0</y>
     <width>1385</width>
     <height>24</height>
    </rect>
   </property>
  </widget>
  <widget class="QStatusBar" name="statusBar"/>
 </widget>
 <customwidgets>
  <customwidget>
   <class>Canvas</class>
   <extends>QOpenGLWidget</extends>
   <header>views/canvas.h</header>
  </customwidget>
 </customwidgets>
 <resources>
  <include location="../resources.qrc"/>
 </resources>
 <connections/>
</ui>
